src/libaktualizr/uptane/imagerepository.cc

```diff
@@ -44,7 +44,7 @@ void ImageRepository::fetchSnapshot(INvStorage& storage, const IMetadataFetcher&
 
   if (local_version > remote_version) {
     throw Uptane::SecurityException(RepositoryType::IMAGE, "Rollback attempt");
-  } else if (local_version < remote_version) {
+  } else {
```

Comment:
I preferred the logic where we compared signatures, because this means that we will store the same thing repeatedly if it hasn't changed. That causes unnecessary writes to the database that I'd like to avoid.

Reply:
It seems like the Timestamp is being treated differently than Snapshot and Targets. I can imagine there's good reason for it, but it's not clear to me. What's the reasoning there?

Reply:
The additional condition `|| timestamp_stored_signature != timestamp.signature()` helps to avoid rewriting the same timestamp metadata in DB each time timestamp metadata are downloaded and checked at the beginning of each update cycle.
This additional condition doesn't bring much benefit in the case of snapshot and targets metadata because these metadata are downloaded only if timestamp metadata are changed (verifySnapshot()/verifyTargets() throws exception, otherwise `fetch_snapshot/fetch_targets` are set to false).
This is very unlikely or can happen very rarely when timestamp changes while snapshot&targets are not, therefore the given approach, helps to avoid unnecessary writes to DB in majority cases.

Reply:
Yes, you make a good point. I do kinda wish these things were made more clear in the code or with comments, though. And I still think low-level tests are important.